integrations/logging/splunk/cluster-output.yaml

```
@@ -18,3 +18,8 @@ spec:
     index: main
     format:
       type: json
+    buffer:
+      timekey: 10s
+      timekey_wait: 1s
+      retry_exponential_backoff_base: 1.5
+      retry_max_interval: 30s
```

Can we add some comments or links to where these parameters are explained?

---

But there are not any comments for any of the other parameters.  Can't we just assume that anyone looking at this would be able to find the documentation for it?  Isn't that what Google is for?

---

well, I just asked because they are optional parameters that we are modifying to get a specific behavior in our integration tests. Of course, we can google the information, but it's really helpful having it written down and a reference for knowing why they shouldn't be changed.

For example, this happened to me when I was trying to understand the "fencing" that we do the database in one of our integration tests, it's really nice but for someone that is not familiar with it (like me) I had to google it. Next time someone sees the same for the first time or after some time, the same question will arise, so, if having a link makes things easier why not adding them?
BTW, I added a link for this in https://github.com/kubearchive/kubearchive/blob/main/test/integration/integration_test.go#L91-L91

The following will be enough for me:
```suggestion
     # Flush and Write times of the logging operator need to be shorter for the integration tests
     # https://kube-logging.dev/4.7/docs/configuration/plugins/outputs/buffer/?highlight=timekey#buffer-timekey
      timekey: 10s 
      timekey_wait: 1s
      retry_exponential_backoff_base: 1.5
      retry_max_interval: 30s
```

---

We've had documentation discussions before and I think we both know how each of us feels about it.  I don't care that much, we can add it.

But why not add these as well?

https://kube-logging.dev/4.7/docs/configuration/plugins/outputs/buffer/?highlight=timekey#buffer-retry_exponential_backoff_base
https://kube-logging.dev/4.7/docs/configuration/plugins/outputs/buffer/?highlight=timekey#buffer-retry_max_interval

We can add them all.

I don't see any difference between adding these and adding the ones for the logging test. Each of these options is there for some reason. Removing them may have adverse effects.

We could add links and pointers for everything. Eventually they become noise and possibly out-of-date. Where do you draw the line?

---

I guess I'm driven by the "feeling" of thinking that I will need this information in the future and I will appreciate having it written in some place and it's easier to find here than elsewhere.

For the "why not adding all the links" once you have one link it's the same place for the others. We can just put https://kube-logging.dev/4.7/docs/configuration/plugins/outputs/buffer/

and it's good enough for me